samples/Program.cs

```diff
@@ -223,5 +230,74 @@ private static void GetContactsExample(EsendexCredentials credentials)
                 Console.Write(ex.Message);
             }
         }
+
+        private static void GetGroupsExample(EsendexCredentials credentials)
+        {
+            var groupService = new GroupService(credentials);
+
+            try
+            {
+                var collection = groupService.GetGroups(_accountReference, PageIndex, PageSize);
+
+                foreach (var item in collection.Groups)
+                {
+                    Console.WriteLine("\tGroup Id:{0}\tName:{1}", item.Id, item.Name);
+                }
+            }
+            catch (WebException ex)
+            {
+                Console.Write(ex.Message);
+            }
+        }
+
+        private static void GetContactsByGroupExample(EsendexCredentials credentials)
+        {
+            var groupService = new GroupService(credentials);
+
+            try
+            {
+                var collection = groupService.GetGroups(_accountReference, PageIndex, PageSize);
+                var contacts = new PagedContactCollection();
+
+                var groupId = "";
+
+                foreach (var item in collection.Groups.Where(item => item.Name == "Test group"))

```

Comment:
As the group name is unique you can use FirstOrDefault with null check to verify if a group has been found.

    var contactGroup = collection.Groups.FirstOrDefault(x => x.Name == "Test group");

    if (contactGroup == null)
       return;